File: Low Level Design (LLD)/DesignPatterns/3. Structural/Mediator/Java/ChatRoom.java
import java.util.ArrayList;
import java.util.List;

public class ChatRoom implements ChatMediator {
    private List<User> users = new ArrayList<>();

    @Override
    public void showMessage(String message, User sender) {
        for (User user : users) {
            // Don't send message to the sender
            if (user != sender) {
                user.receive(message, sender);
            }
        }
    }

    @Override
    public void addUser(User user) {
        users.add(user);
    }
}

File: Low Level Design (LLD)/DesignPatterns/3. Structural/Mediator/Cpp/src/ChatRoom.cpp
#include "../include/ChatRoom.h"
#include <iostream>

void ChatRoom::showMessage(const std::string& message, User* sender) {
    for (auto user : users) {
        if (user != sender) {
            user->receive(message, sender);
        }
    }
}

void ChatRoom::addUser(User* user) {
    users.push_back(user);
}
